piff/hyperpsf.py

```diff
@@ -0,0 +1,277 @@
+                                            data vectors (produced by model for each star).
+        """
+        self.model = model
+        self.interp = interp
+
+        self.n_iter = 0
+        self._p_hist = []
+
+    @classmethod
+    def build(cls, stars, model, interp, logger=None, **fit_kwargs):
+        """The main driver function to build an HyperPSF model from data.
+
+        :param stars:                       A list of StarData instances.
+        :param model:                       A Model instance that defines how to model the individual PSFs
+                                            at the location of each star.
+        :param interp:                      An Interp instance that defines how to do the interpolation of the
+                                            data vectors (produced by model for each star).
+        :param fit_kwargs:                  kwargs to pass to iminuit fitter
+        :param logger:                      A logger object for logging debug info. [default: None]
+
+        :returns: an HyperPSF instance
+
+        http://iminuit.readthedocs.io/en/latest/api.html
+        """
+        if logger:
+            logger.info("Start building HyperPSF using %s stars", len(stars))
+            logger.debug("Model is %s", model)
+            logger.debug("Interp is %s", interp)
+
+        psf = cls(model, interp)
+        if logger:
+            logger.info("Fitting PSF model")
+        psf.fit(stars, logger=logger, **fit_kwargs)
+        if logger:
+            logger.debug("Done building PSF")
+        return psf
+
+    def fit(self, stars, model_keys=[], interp_keys=[], model_comparer=None, model_comparer_weights=None, model_init=None, interp_init=None, model_error=None, interp_error=None, model_limit=None, interp_limit=None, logger=None, skip_fit=False, **kwargs):
+        """Fit the model!
+        :param stars:                       A list of StarData instances.
+        :param model:                       A Model instance that defines how to model the individual PSFs
+                                            at the location of each star.
+        :param interp:                      An Interp instance that defines how to do the interpolation of the
+                                            data vectors (produced by model for each star).
+        :param model_keys, interp_keys:     A list that gives the attributes of model/interp that the params correspond to
+        :param model_comparer:              Model that gets applied to both stars and model_stars to compare success of model. If none is given, compare pixels.
+        :param model_comparer_weights:      Weights to apply to the params from model_comparer
+        :param model_init, interp_init:     A list that gives the initial values
+        :param model_error, interp_error:   A list that gives the initial step sizes
+        :param model_limit, interp_limit:   A list that gives the limits of the fit
```

Comment:
That's fine.  If you want to comment on anything in #20, please do.  Maybe we can plan to merge that this week sometime, and then see if this all makes sense with that code flow.